mypy/server/astmerge.py

```diff
@@ -429,6 +432,10 @@ def visit_union_type(self, typ: UnionType) -> None:
         for item in typ.items:
             item.accept(self)
 
+    def visit_placeholder_type(self, t: PlaceholderType) -> None:
```

Comment:
I think this also should raise, since this should be called after semantic analysis is done. But maybe I am wrong because there are many visit methods for synthetic types. Or maybe it was mistakenly made `SyntheticTypeVisitor` instead of just `TypeVisitor`? Anyway, this is not important for this PR.

Reply:
I think it was intentional that this visitor is a SyntheticTypeVisitor -- we apparently try running this visitor on `FuncBase.unanalyzed_type` in the `process_base_func` method up above. Not entirely sure why, but I figured it'd be safest to just leave this alone.